### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by Estrella1024** [UoA-eResearch/leapmotion_gestures#7]
Excuse me, how is your model trained?

**Comment by Estrella1024** [comment]
And What are the advantages of using the LSTM deep learning model in combination with leap motion equipment?

**Comment by neon-ninja** [comment]
Hi - training data is recorded to CSVs with https://github.com/UoA-eResearch/leapmotion_gestures/blob/master/record.py. The model is trained on that data in https://github.com/UoA-eResearch/leapmotion_gestures/blob/master/exploration-many2one.ipynb. The advantage of using an LSTM is that it can recognise gestures that are defined by a small amount of movement - such as the "driving" gesture. Or another example would be detecting the difference between the "stop" gesture and the "wave" gesture. @aleathwick would know more.

**Comment by aleathwick** [comment]
Hi there, as @neon-ninja has said, code for model training can be found in the Jupyter notebook [https://github.com/UoA-eResearch/leapmotion_gestures/blob/master/exploration-many2one.ipynb](https://github.com/UoA-eResearch/leapmotion_gestures/blob/master/exploration-many2one.ipynb). Training data is limited to what we could record with a few volunteers, meaning overly complex models would tend to overfit and not perform well on new unseen data. As a result, the models are not large and can be trained in a matter of minutes on a CPU, but nevertheless perform very well.

> What are the advantages of using the LSTM deep learning model in combination with leap motion equipment?

Using deep learning models with leap motion equipment opens the door to cool applications like recognizing hand gestures, as we have done here. Why specifically LSTMs? Leap motion devices report measurements many times each second, producing an ordered sequence of data points (i.e., the data is sequential in nature). LSTMs are designed for use with sequential data, processing each timestep in order and learning to store and use relevant information from previous timesteps. For example, the 'stop' gesture is just like the 'wave' gesture if you look at a single snapshot, but if you look at it over time, the 'wave' gesture adds side to side movement, which the LSTM can pick up and use to differentiate the two gestures. LSTMs perform much better on sequential data than 'ordinary' fully connected feed forward models, which are not designed for sequential data.

I hope that that answers your questions, let me know if you would like to know any more or need clarification.

**Comment by Estrella1024** [comment]
> Hi - training data is recorded to CSVs with https://github.com/UoA-eResearch/leapmotion_gestures/blob/master/record.py. The model is trained on that data in https://github.com/UoA-eResearch/leapmotion_gestures/blob/master/exploration-many2one.ipynb. The advantage of using an LSTM is that it can recognise gestures that are defined by a small amount of movement - such as the "driving" gesture. Or another example would be detecting the difference between the "stop" gesture and the "wave" gesture. @aleathwick would know more.

Thank you so much!

**Comment by Estrella1024** [comment]
> Hi there, as @neon-ninja has said, code for model training can be found in the Jupyter notebook https://github.com/UoA-eResearch/leapmotion_gestures/blob/master/exploration-many2one.ipynb. Training data is limited to what we could record with a few volunteers, meaning overly complex models would tend to overfit and not perform well on new unseen data. As a result, the models are not large and can be trained in a matter of minutes on a CPU, but nevertheless perform very well.
> 
> > What are the advantages of using the LSTM deep learning model in combination with leap motion equipment?
> 
> Using deep learning models with leap motion equipment opens the door to cool applications like recognizing hand gestures, as we have done here. Why specifically LSTMs? Leap motion devices report measurements many times each second, producing an ordered sequence of data points (i.e., the data is sequential in nature). LSTMs are designed for use with sequential data, processing each timestep in order and learning to store and use relevant information from previous timesteps. For example, the 'stop' gesture is just like the 'wave' gesture if you look at a single snapshot, but if you look at it over time, the 'wave' gesture adds side to side movement, which the LSTM can pick up and use to differentiate the two gestures. LSTMs perform much better on sequential data than 'ordinary' fully connected feed forward models, which are not designed for sequential data.
> 
> I hope that that answers your questions, let me know if you would like to know any more or need clarification.

Thank you so much. If I have any questions, I will consult with you.
